### PR TITLE
First modification for pep237 with pydocstringformatter 

### PR DESCRIPTION
**Pull request by Pierre-Sassoulas**
## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :scroll: Docs          |

## Related Issue

Some results are not correct, this is the occasion to make pydocstringformatter better and to add astroid to its primer. We will handle the incorrect one in #1796

### CODE REVIEW

File: astroid/brain/brain_boto3.py [hunk 1]
```diff
@@ -2,7 +2,7 @@
 # For details: https://github.com/PyCQA/astroid/blob/main/LICENSE
 # Copyright (c) https://github.com/PyCQA/astroid/blob/main/CONTRIBUTORS.txt
 
-"""Astroid hooks for understanding boto3.ServiceRequest()"""
+"""Astroid hooks for understanding boto3.ServiceRequest()."""
```
Comment (Pierre-Sassoulas) [inline astroid/brain/brain_boto3.py:5]:
```suggestion
"""Astroid hooks for understanding ``boto3.ServiceRequest()``."""
```

File: astroid/brain/brain_collections.py [hunk 2]
```diff
@@ -83,8 +83,10 @@ def __class_getitem__(cls, item): return cls"""
 
 def _looks_like_subscriptable(node: ClassDef) -> bool:
     """
-    Returns True if the node corresponds to a ClassDef of the Collections.abc module that
-    supports subscripting
+    Returns True if the node corresponds to a ClassDef of the Collections.abc module
+    that.
+
+    supports subscripting.
```
Comment (Pierre-Sassoulas) [inline astroid/brain/brain_collections.py:89]:
```suggestion
    Returns True if the node corresponds to a ClassDef of the Collections.abc module
    that supports subscripting.
```

File: astroid/brain/brain_unittest.py [hunk 3]
```diff
@@ -11,8 +11,12 @@
...
     """
-    In the unittest package, the IsolatedAsyncioTestCase class is imported lazily, i.e only
-    when the __getattr__ method of the unittest module is called with 'IsolatedAsyncioTestCase' as
+    In the unittest package, the IsolatedAsyncioTestCase class is imported lazily,
+    i.e only.
+
+    when the __getattr__ method of the unittest module is called with
+    'IsolatedAsyncioTestCase' as.
+
     argument. Thus the IsolatedAsyncioTestCase is not imported statically (during import time).
```
Comment (Pierre-Sassoulas) [inline astroid/brain/brain_unittest.py:20]:
```suggestion
    In the unittest package, the IsolatedAsyncioTestCase class is imported lazily.
    
    I.E. only when the ``__getattr__`` method of the unittest module is called with
    'IsolatedAsyncioTestCase' as argument. Thus the IsolatedAsyncioTestCase
    is not imported statically (during import time).
```

File: astroid/context.py [hunk 4]
```diff
@@ -49,7 +49,7 @@ def __init__(self, path=None, nodes_inferred=None):
             self._nodes_inferred = nodes_inferred
         self.path = path or set()
         """
-        :type: set(tuple(NodeNG, optional(str)))
+        :type: set(tuple(NodeNG, optional(str))).
```
Comment (Pierre-Sassoulas) [inline astroid/context.py:52]: Hmm.
```suggestion
        :type: set(tuple(NodeNG, optional(str)))
```

File: astroid/inference.py [hunk 5]
```diff
@@ -971,7 +972,8 @@ def _do_compare(
 ) -> bool | type[util.Uninferable]:
     """
     If all possible combinations are either True or False, return that:
-    >>> _do_compare([1, 2], '<=', [3, 4])
+    >>> _do_compare([1, 2], '<=', [3, 4]).
```
Comment (Pierre-Sassoulas) [inline astroid/inference.py:975]: It will be fixed in https://github.com/DanielNoord/pydocstringformatter/issues/168

File: astroid/modutils.py [hunk 6]
```diff
@@ -339,8 +339,9 @@ def file_info_from_modpath(
     path: Sequence[str] | None = None,
     context_file: str | None = None,
 ) -> spec.ModuleSpec:
-    """given a mod path (i.e. split module / package name), return the
-    corresponding file, giving priority to source file over precompiled
+    """Given a mod path (i.e. split module / package name), return the
+    corresponding file, giving priority to source file over precompiled.
+
     file if it exists
```
Comment (Pierre-Sassoulas) [inline astroid/modutils.py:345]:
```suggestion
    """Given a mod path (i.e. split module / package name), return the
    corresponding file.
    
    Giving priority to source file over precompiled file if it exists.
```

**Comment by DanielNoord** [comment]
@Pierre-Sassoulas Will you open issues about stuff that is broken?

**Comment by Pierre-Sassoulas** [comment]
> Will you open issues about stuff that is broken?

Yes, if you think it's necessary : https://github.com/PyCQA/astroid/pull/1792#discussion_r973672856 for example would be hard to fix imo ? 

I think the only clear cut one would be : https://github.com/PyCQA/astroid/pull/1792#discussion_r973673605 (only typing information without text)

**Comment by DanielNoord** [comment]
> Yes, if you think it's necessary : [#1792 (comment)](https://github.com/PyCQA/astroid/pull/1792#discussion_r973672856) for example would be hard to fix imo ?

https://github.com/DanielNoord/pydocstringformatter/issues/127

> I think the only clear cut one would be : [#1792 (comment)](https://github.com/PyCQA/astroid/pull/1792#discussion_r973673605) (only typing information without text)

I think this can also be fixed. We just need to wrap>split>period instead of split>wrap>period.
Perhaps even wrap>split>wrap>period.

Something to investigate at least!

**Comment by Pierre-Sassoulas** [comment]
I created https://github.com/DanielNoord/pydocstringformatter/issues/175

I think we can"t add pydocstringformatter right now, but we might want to integrate the change that make sense right now, what do you think ?

**Comment by DanielNoord** [comment]
> I think we can"t add pydocstringformatter right now, but we might want to integrate the change that make sense right now, what do you think ?

Agreed! Is this reviewable?

**Comment by Pierre-Sassoulas** [comment]
> Agreed! Is this reviewable?

Not yet, I'll clean this up in the coming week

**Comment by Pierre-Sassoulas** [comment]
This is reviewable now :)

File: astroid/astroid_manager.py [hunk 7]
```diff
@@ -1,5 +1,7 @@
 """
-This file contain the global astroid MANAGER, to prevent circular import that happened
+This file contain the global astroid MANAGER, to prevent circular import that
+happened.
```
Comment (DanielNoord) [inline astroid/astroid_manager.py:3]: This sentence is now broken.

File: astroid/brain/brain_hypothesis.py [hunk 8]
```diff
@@ -39,7 +38,8 @@ def is_decorated_with_st_composite(node):
 
 def remove_draw_parameter_from_composite_strategy(node):
     """Given that the FunctionDef is decorated with @st.composite, remove the
-    first argument (`draw`) - it's always supplied by Hypothesis so we don't
+    first argument (`draw`) - it's always supplied by Hypothesis so we don't.
```
Comment (DanielNoord) [inline astroid/brain/brain_hypothesis.py:41]: This is broken.

File: astroid/exceptions.py [hunk 9]
```diff
@@ -399,7 +400,9 @@ def __init__(self, target: nodes.NodeNG) -> None:
 
 
 class StatementMissing(ParentMissingError):
-    """Raised when a call to node.statement() does not return a node. This is because
+    """Raised when a call to node.statement() does not return a node.
+
+    This is because
```
Comment (DanielNoord) [inline astroid/exceptions.py:405]: L406 should probably be appended here.

File: astroid/manager.py [hunk 10]
```diff
@@ -2,8 +2,9 @@
 # For details: https://github.com/PyCQA/astroid/blob/main/LICENSE
 # Copyright (c) https://github.com/PyCQA/astroid/blob/main/CONTRIBUTORS.txt
 
-"""astroid manager: avoid multiple astroid build of a same module when
-possible by providing a class responsible to get astroid representation
+"""Astroid manager: avoid multiple astroid build of a same module when
+possible by providing a class responsible to get astroid representation.
+
 from various source and using a cache of built modules)
```
Comment (DanielNoord) [inline astroid/manager.py:8]: This is broken. Also note there is a single `)` without a `(`

File: astroid/modutils.py [hunk 11]
```diff
@@ -339,9 +339,10 @@ def file_info_from_modpath(
     path: Sequence[str] | None = None,
     context_file: str | None = None,
 ) -> spec.ModuleSpec:
-    """given a mod path (i.e. split module / package name), return the
-    corresponding file, giving priority to source file over precompiled
-    file if it exists
+    """Given a mod path (i.e. split module / package name), return the
+    corresponding file.
+
+    Giving priority to source file over precompiled file if it exists.
```
Comment (DanielNoord) [inline astroid/modutils.py:345]: Can't this be put on the summary? It's part of the main sentence

File: astroid/modutils.py [hunk 12]
```diff
@@ -481,8 +481,9 @@ def get_module_files(
 
 
 def get_source_file(filename: str, include_no_ext: bool = False) -> str:
-    """given a python module's file name return the matching source file
-    name (the filename will be returned identically if it's already an
+    """Given a python module's file name return the matching source file
+    name (the filename will be returned identically if it's already an.
```
Comment (DanielNoord) [inline astroid/modutils.py:485]: This is broken

File: astroid/nodes/as_string.py [hunk 13]
```diff
@@ -51,23 +51,24 @@ def _docs_dedent(self, doc_node: Const | None) -> str:
...
+        """Wrap child in parens only if required to keep same semantics."""
         if self._should_wrap(node, child, is_left):
             return f"({child.accept(self)})"
 
         return child.accept(self)
 
     def _should_wrap(self, node, child, is_left):
         """Wrap child if:
-        - it has lower precedence
+        - it has lower precedence.
```
Comment (DanielNoord) [inline astroid/nodes/as_string.py:70]: This should probably be done differently.

File: astroid/nodes/node_classes.py [hunk 14]
```diff
@@ -2,7 +2,10 @@
 # For details: https://github.com/PyCQA/astroid/blob/main/LICENSE
 # Copyright (c) https://github.com/PyCQA/astroid/blob/main/CONTRIBUTORS.txt
 
-"""Module for some node classes. More nodes in scoped_nodes.py"""
+"""Module for some node classes.
+
+More nodes in scoped_nodes.py
```
Comment (DanielNoord) [inline astroid/nodes/node_classes.py:7]: This is nog longer true. It is now a package.

File: astroid/nodes/scoped_nodes/mixin.py [hunk 15]
```diff
@@ -19,8 +19,9 @@
 
 
 class LocalsDictNodeNG(node_classes.LookupMixIn):
-    """this class provides locals handling common to Module, FunctionDef
-    and ClassDef nodes, including a dict like interface for direct access
+    """This class provides locals handling common to Module, FunctionDef
+    and ClassDef nodes, including a dict like interface for direct access.
```
Comment (DanielNoord) [inline astroid/nodes/scoped_nodes/mixin.py:23]: This is broken.

File: astroid/nodes/scoped_nodes/scoped_nodes.py [hunk 16]
```diff
@@ -691,6 +692,7 @@ def __init__(
     ):
         """
         :param lineno: The line that this node appears on in the source code.
+
```
Comment (DanielNoord) [inline astroid/nodes/scoped_nodes/scoped_nodes.py:695]: I think this breaks valid Sphinx documentation

File: astroid/raw_building.py [hunk 17]
```diff
@@ -349,8 +349,9 @@ def inspect_build(
         modname: str | None = None,
         path: str | None = None,
     ) -> nodes.Module:
-        """build astroid from a living module (i.e. using inspect)
-        this is used when there is no python source code available (either
+        """Build astroid from a living module (i.e. using inspect)
+        this is used when there is no python source code available (either.
```
Comment (DanielNoord) [inline astroid/raw_building.py:353]: This is broken

File: astroid/astroid_manager.py (hunk 7 above)
Comment (Pierre-Sassoulas) [reply astroid/astroid_manager.py:3]: I missed so much of those that I'll need to redo the commit entirely. I'll get to it. There's a lot of change to review.

**Comment by Pierre-Sassoulas** [comment]
Arg, and there's also conflicts now. Damn.

**Review by DanielNoord — APPROVED** [review]

**Pull request merged**